cmd/osbuild-worker/jobimpl-ostree-resolve.go

```diff
@@ -17,6 +17,10 @@ type OSTreeResolveJobImpl struct {
 }
 
 func (job *OSTreeResolveJobImpl) CompareBaseURL(baseURLStr string) (bool, error) {
+	if job.RepositoryMTLSConfig == nil {
+		return false, nil
+	}
```

Comment:
Should probably also check if BaseURL is nil.
```suggestion
	if job.RepositoryMTLSConfig == nil || job.RepositoryMTLSConfig.BaseURL == nil {
		return false, nil
	}
```